### PR TITLE
copr: vectorize `conv`.

### PR DESCRIPTION
**Pull request by cireu**
Signed-off-by: Zhu Zihao <all_but_last@163.com>

PCP #5751 

###  What have you changed?

Vectorize `conv`
Clean the implementation of non-vectorized one.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?

No
###  Refer to a related PR or issue link (optional)

Finish https://github.com/tikv/tikv/pull/5921

### CODE REVIEW

**Comment by cireu** [comment]
@sre-bot  /run-integration-copr-test copr-test=pr/50

File: components/tidb_query/src/expr_util/conv.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+impl IntWithSign {
+    fn from_int(num: Int) -> IntWithSign {
+        IntWithSign(num.wrapping_abs() as u64, num < 0)
+    }
+
+    fn from_signed_uint(num: u64, is_neg: bool) -> IntWithSign {
+        IntWithSign(num, is_neg)
+    }
+
+    // Shrink NUM to fit the boundary of i64.
```
Comment (mapleFU) [inline components/tidb_query/src/expr_util/conv.rs]: maybe NUM --> `num` is better?

File: components/tidb_query/src/expr_util/conv.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    fn shrink_from_signed_uint(num: u64, is_neg: bool) -> IntWithSign {
+        let value = if is_neg {
+            num.min(-Int::min_value() as u64)
+        } else {
+            num.min(Int::max_value() as u64)
+        };
+        IntWithSign::from_signed_uint(value, is_neg)
+    }
+
+    fn format_radix(mut x: u64, radix: u32) -> String {
```
Comment (mapleFU) [inline components/tidb_query/src/expr_util/conv.rs]: would you mind adding `assert(is_valid_base(radix))` here?

File: components/tidb_query/src/expr_util/conv.rs (hunk 1 above)
Comment (cireu) [reply components/tidb_query/src/expr_util/conv.rs]: Ok, I didn't find some rules to guide this in CODE_COMMENT_STYLE.md. So I follow the convention in some Lisp project -- using UPPER CASE to identify the argument.

File: components/tidb_query/src/expr_util/conv.rs (hunk 2 above)
Comment (cireu) [reply components/tidb_query/src/expr_util/conv.rs]: What about `debug_assert!` ?

Comment (cireu) [reply components/tidb_query/src/expr_util/conv.rs]: ping @mapleFU

Comment (mapleFU) [reply components/tidb_query/src/expr_util/conv.rs]: pong

Comment (mapleFU) [reply components/tidb_query/src/expr_util/conv.rs]: Yes, `debug_assert` is ok here.

Comment (cireu) [reply components/tidb_query/src/expr_util/conv.rs]: I'm sorry, I want to commit this but it was merged

**Pull request merged**